google/datalab/contrib/pipeline/composer/_composer.py

```diff
@@ -29,22 +30,34 @@ def __init__(self, zone, environment):
     """
     self._zone = zone
     self._environment = environment
+    self._gcs_dag_location = None
 
   def deploy(self, name, dag_string):
     client = gcs.Client()
-    bucket = client.get_bucket(self.bucket_name)
-    filename = 'dags/{0}.py'.format(name)
+    try:
+      gcs_dag_location_splits = self.gcs_dag_location.split('/')
+      bucket_name = gcs_dag_location_splits[2]
+      # Usually the splits are like ['gs:', '', 'foo_bucket', 'dags']. But we could have additional
+      # parts after the bucket. In those cases, the final file path needs to include those as well
+      additional_parts = ''
+      if len(gcs_dag_location_splits) > 4:
+        additional_parts = '/' + '/'.join(gcs_dag_location_splits[4:])
+      filename = self.gcs_dag_location.split('/')[3] + additional_parts + '/{0}.py'.format(name)
+    except (AttributeError, IndexError):
+      raise ValueError('Error in dag location from Composer environment {0}'.format(
+        self._environment))
+
+    bucket = client.get_bucket(bucket_name)
     blob = gcs.Blob(filename, bucket)
     blob.upload_from_string(dag_string)
 
   @property
-  def bucket_name(self):
-    # TODO(rajivpb): Get this programmatically from the Composer API
-    return 'airflow-staging-test36490808-bucket'
-
-  @property
-  def get_bucket_name(self):
-    # environment_details = Api().environment_details_get(self._zone, self._environment)
-
-    # TODO(rajivpb): Get this programmatically from the Composer API
-    return 'airflow-staging-test36490808-bucket'
+  def gcs_dag_location(self):
+    if not self._gcs_dag_location:
+      environment_details = Api.environment_details_get(self._zone, self._environment)
+      if 'config' not in environment_details \
```

Comment:
Per style guide, don't use \ for line continuation.  Wrap the statement in parens instead.

Reply:
Fixed.